frontend/src/util/MarkerManager.js

```diff
@@ -7,20 +7,38 @@ class MarkerManager {
   }
 
   updateMarkers(restaurants) {
+    const restaurantsObject = {};
+
+    restaurants.forEach((restaurant) => {
+      restaurantsObject[restaurant.id] = restaurant;
+    });
+
+    Object.keys(this.markers).forEach((key) => {
+      const marker = this.markers[key];
+      if (!restaurantsObject[marker.id]) {
+        delete this.markers[marker.id];
```

Comment:
This is that one bug... you need to use `key` instead of `marker.id` for these lines (since markers don't have ids)